### PR TITLE
chore(deps): update type checkers

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by mr-cal — APPROVED** [review]

File: craft_parts/parts.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def get_scriptlet(self, step: Step) -> Optional[str]:
             return self.override_build
         if step == Step.STAGE:
             return self.override_stage
-        if step == step.PRIME:
+        if step == Step.PRIME:
```
Comment (cmatsuoka) [inline craft_parts/parts.py:158]: This is an interesting bug, how did linters let this pass?

File: craft_parts/plugins/rust_plugin.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def validate_environment(
                     plugin_name="rust",
                     part_dependencies=part_dependencies,
                 )
-            options.rust_channel = "none"
+            options.rust_channel = "none"  # type: ignore[misc]
```
Comment (cmatsuoka) [inline craft_parts/plugins/rust_plugin.py:102]: The type is already optional string, why is this necessary?

**Review by cmatsuoka — APPROVED** [review]

File: craft_parts/parts.py (hunk 1 above)
Comment (lengau) [reply craft_parts/parts.py:158]: I'm not entirely sure — I guess because functionally it wasn't a bug (`<instance>.PRIME == <class>.PRIME`, though strictly `<instance>.PRIME is <class>.PRIME.value`)?

File: craft_parts/plugins/rust_plugin.py (hunk 2 above)
Comment (lengau) [reply craft_parts/plugins/rust_plugin.py:107]: When I wrote this last week, I thought we had tested the rust plugin to include this, but looking again we haven't.

The complaint was:

```
craft_parts/plugins/rust_plugin.py:103: error: Property "rust_channel" defined in "RustPluginProperties" is read-only  [misc]
```

So this is actually a perfect example of mypy uncovering a bug for us! :tada: 

I've created #559 for that and updated the file with an explanatory comment and a note about removing this ignore comment once we fix it.
